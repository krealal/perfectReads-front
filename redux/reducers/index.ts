import { combineReducers } from "redux";
import reviewReducer from "./reviewReducer";
import userReducer from "./userReducer";

const rootReducer = combineReducers({
  reviewsList: reviewReducer,
  user: userReducer,
});

export default rootReducer;
