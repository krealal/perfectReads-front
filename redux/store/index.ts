import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import { createStore, applyMiddleware, Store } from "redux";
import rootReducer from "../reducers";

import { Context, createWrapper } from "next-redux-wrapper";

const store = (context: Context) =>
  createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

export const wrapper = createWrapper<Store>(store);
