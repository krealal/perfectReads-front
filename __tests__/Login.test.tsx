import { render, screen } from "@testing-library/react";
import { fireEvent } from "@testing-library/dom";
import Login from "../pages/login";
import { wrapper } from "../redux/store";
import TestRenderer from "react-test-renderer";

describe("given a login page", () => {
  describe("when its rendered", () => {
    test("then it should find a textbox and a button in the document", async () => {
      const WrappedComponent = await wrapper.withRedux(Login);
      render(<WrappedComponent></WrappedComponent>);

      const expectedInput = "textbox";
      const expectedButton = "button";

      const expectedOutput = screen.getByRole(expectedInput, {
        name: /name/i,
      });
      const expectedOutputTwo = screen.getByRole(expectedButton);

      expect(expectedOutput).toBeInTheDocument();
      expect(expectedOutputTwo).toBeInTheDocument();
    });
  });

  describe("when its rendered and clicks on button login", () => {
    test("then it should call router and push", async () => {
      const WrappedComponent = wrapper.withRedux(Login);

      render(<WrappedComponent />);

      const button = screen.getByRole("button", { name: /login/i });
      await fireEvent.click(button);
    });
  });

  describe("when its rendered", () => {
    test("then it should make loginPage into json and compare with snapshot", async () => {
      const WrappedComponent = wrapper.withRedux(Login);

      const loginPage = await TestRenderer.create(<WrappedComponent />);

      expect(loginPage.toJSON()).toMatchSnapshot();
    });
  });
});
