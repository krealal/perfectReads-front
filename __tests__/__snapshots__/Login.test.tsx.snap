// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given a login page when its rendered then it should make loginPage into json and compare with snapshot 1`] = `
<div
  className="sc-dkPtRN dmjCiJ"
>
  <div
    className="tittleContainer"
  >
    <h1
      className="tittlePerfect"
    >
      perfect
    </h1>
    <h1
      className="tittleReads"
    >
      reads
    </h1>
  </div>
  <section
    className="sectionForm"
  >
    <form
      autoComplete="off"
      className="MuiBox-root css-1m8tcw4"
      noValidate={true}
      onSubmit={[Function]}
    >
      <h2
        className="sc-gsDKAQ eotxPF"
      >
        Log in
      </h2>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="username"
          id="username-label"
        >
          Username
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Username
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          password
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="sc-bdvvtL gzBWRV"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={true}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          LOGIN
        </button>
        <p>
          Â¿You need an account? 
          <a
            href="/register"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Sign Up
          </a>
        </p>
      </div>
    </form>
  </section>
  <img
    alt="library"
    className="libraryImage"
    src="https://a.travel-assets.com/findyours-php/viewfinder/images/res40/200000/200441.jpg"
  />
</div>
`;
